### PR TITLE
allow for setting middlewares via config and via Filament Resources

### PR DESCRIPTION
**Pull request by eelco2k**

### CODE REVIEW

**Comment by rupadana** [comment]
i think its better to use ApiServicePlugin::make()->middleware() cause filament had filament API in Panel Provider too, so the dev can focus on the Provider for adding the middleware

**Comment by eelco2k** [comment]
@rupadana i've refactored it. you are right. this way a dev can use middlewares per panel instance.

File: README.md [hunk 1]
```diff
@@ -191,6 +192,36 @@ You can edit prefix & group route name as you want, default this plugin use mode
...
+
+You can add or override middlewares at two specific places. Via the Filament Panel Provider and/or via the Resources $routeMiddleware.
+
+If you set `route.use_resource_middlewares` to true, the package will register the middlewares for that specific resource as defined in:
+
+```php
+class BlogResource extends Resource
+    {
+        ...
+        protected static string | array $routeMiddleware = []; // <-- your specific resource middlewares
```
Comment (rupadana) [inline README.md:205]: please create interface called `HasRouteMiddleware` and there are property $routeMiddleware and getter method `getRouteMiddleware`

Comment (eelco2k) [reply README.md:205]: `protected static string | array $routeMiddleware = [];` is something from filament resource itself.

check filament/filament/src/Resources/Resource.php

Comment (rupadana) [reply README.md:205]: ah i see, i never use it before.

**Comment by rupadana** [comment]
@eelco2k Thank you for your Contribution 👍 . this PR is already merged to `dev`. Please take some test of it, and if you want to contribute again, please create new branch from `dev`. Thank you.

**Pull request merged**